added ansible

joshuacox opened the pull request:
I made an ansible playbook I'm using to maintain my rancher server and it's agents,  I also have some octohost  repos to be used with http://octohost.io 
https://github.com/joshuacox/OctoRancherAgent
https://github.com/joshuacox/OctoRancherServer

But the agent will not work in octohost as is until my pull request goes through over at octohost to add in the ability for it to start images in privileged mode.  It works on mine by upgrading /usr/bin/octo to https://raw.githubusercontent.com/joshuacox/octohost/master/bin/octo

ibuildthecloud commented:
@joshuacox The privileged flag on the rancher/agent is not 100% required.  We add the flag for two reasons.  First, if you have SELinux on and are running on Fedora/CentOS/RHEL, they don't allow you to talk to `/var/run/docker.sock` without `--privileged`.  Second, we add it to let people know that the agent does have full control over the box.  We just want to be explicit that our agent is not an isolated app.

If you upgrade the rancher/server and it requires a new version of rancher/agent, the server will tell the agent to pull the new image and restart the agent.  Will this cause any problems with octohost?  This just means that if you've configured to launch version 0.1 of the agent on boot and the server needs agent 0.2, the agent will start 0.1 and then spawn a new container with 0.1.

joshuacox commented:
@ibuildthecloud That is very impressive that the agent can control so much of the box without the --privileged flag.

So far as octohost, the agent and server can run fine in there, but once the agent starts another agent octo doesn't really know about the second one.  This is fine, you just have to ssh in and manually use docker commands to manipulate from there, or you can use the rancher interface.  There is a bit of a disconnect between how octohost stores configurations and rancher.  This pretty much amounts to if you use rancher to reboot a container managed by octo, you'll lose any customizations that only octo knows about either through consul values, or through magic comments in the Dockerfile.  And Octo does not know about or manipulate the rancher containers.  That being said, rancher seems to live happily inside of an octohost and can spawn containers at will.

Right now I have a pure RancherOS setup so I can go through some the rancher blogposts without being distracted by any non-rancher issues.  But I will continue to test out both the agent and the server with my other dockers setups (generic, octo, and deis at the moment, thought I haven't stuck rancher in deis, that's next).  

I updated the ansible playbook with a few playbooks to get the rancherOS container compatible with ansible (ie switch to ubuntu-console and bootstrap python into it), I might move all this into an option for the packer builder.  I also, added in label support for the agents.
